UOP1_Project/Assets/Scripts/Sound/SoundManager.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Audio;
+
+public class SoundManager : MonoBehaviour
+{
+	public static SoundManager singleton;
+
+	[Header ("Init Variables")]
+	[Tooltip ("Amount of sound emitters created on Start")]
+	public int initialPoolSize = 1;
+
+	[Header ("Mixer Groups")]
+	public AudioMixerGroup masterMixer;
+	public AudioMixerGroup sfxMixer;
+	public AudioMixerGroup musicMixer;
+
+	private List<SoundEmitter> soundEmitterPool = new List<SoundEmitter>();
+	private GameObject soundEmittersContainer;
+	private int poolIndex = 0;
+
+	private void Awake()
+	{
+		DontDestroyOnLoad(this);
+
+		if (!SoundManager.singleton)
+			singleton = this;
+		else
+		{
+			Debug.LogWarning("One sound manager already present in the game, destroying " + this.name + ".");
+			Destroy(this);
+		}
+	}
+
+	void Start()
+    {
+		InitEmitterPool();
+    }
+
+	#region mixer groups functions
+	public static void ChangeVolumeOfMixerGroup(AudioMixerGroup mixer, float newVolumeNormalized)
+	{
+		mixer.audioMixer.SetFloat("Volume", NormalizedToMixerValue(newVolumeNormalized));
+	}
+
+	public static void SaveVolumeOfMixerGroup(AudioMixerGroup mixer)
+	{
+		float tempVolume;
+
+		if (mixer.audioMixer.GetFloat("Volume", out tempVolume))
+		{
+			PlayerPrefs.SetFloat(mixer.name, tempVolume);
+		}
+		else
+			Debug.LogError("Could not save volume for mixer group " + mixer.name + ". It does not contain an exposer variable with name Volume" );
+	}
+
+	public static void LoadVolumeOfMixerGroup(AudioMixerGroup mixer)
+	{
+		float tempVolume = PlayerPrefs.GetFloat(mixer.name, Mathf.Infinity);
+
+		if (tempVolume != Mathf.Infinity)
+		{
+			mixer.audioMixer.SetFloat("Volume", tempVolume);
+		}
+		else
+			Debug.Log("There is no saved volume preferences for mixer " + mixer.name + ", could not load volume.");
+	}
+
+	#region mixerHelpers
+	// Both MixerValueNormalized and NormalizedToMixerValue functions are used for easier transformations when using UI sliders normalized format
+	private static float MixerValueNormalized(float value)
+	{
+		return  (-(value - 80) / 80) - 1;
+	}
+	private static float NormalizedToMixerValue(float normalizedValue)
+	{
+		return -80 + (normalizedValue * 80);
+	}
+	#endregion
+	#endregion
+
+	#region play sounds functions
+	// Vector3.zero is used as a marker of a 2D sound
+	public SoundEmitter Play2DSound(AudioClip clip, AudioMixerGroup mixer, bool loop = false, float volume = 1, float pitch = 1, float spartialBlend = 0)
```

Comment:
spatialBlend does not need to be a parameter for 2D sounds. Like you did in your previous version this is always 0 if i am not mistaken.

(also you misspelled it :) )

Reply:
woops, trying to manage too many things at one atm, fixed it.